searx/engines/__init__.py

```diff
@@ -219,18 +218,6 @@ def set_language_attributes(engine: Engine):
     # language_support
     engine.language_support = len(engine.supported_languages) > 0
 
-    # assign language fetching method if auxiliary method exists
-    if hasattr(engine, '_fetch_supported_languages'):
-        headers = {
-            'User-Agent': gen_useragent(),
-            'Accept-Language': "en-US,en;q=0.5",  # bing needs to set the English language
-        }
-        engine.fetch_supported_languages = (
-            # pylint: disable=protected-access
-            lambda: engine._fetch_supported_languages(get(engine.supported_languages_url, headers=headers))
-        )
```

Comment:
@MarcAbonce I moved the closures (lambda) from your commit 9b6ffed06 to the `searxng_extra/update/update_languages.py` .. IMO these manipulation of engine's namespace is not needed here when initialize the engine (what happens e.g. when a SearXNG service starts).

The lambda function is only called in the `update_languages.py`, that's why I think it is better and more clear when we move this part to the `update_languages.py`.

What do you think, did I missed something?

Reply:
I agree, this code could be in `update_languages.py` (so the comment `# bing needs to set the English language` has the proper context).